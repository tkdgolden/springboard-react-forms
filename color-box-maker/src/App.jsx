import { useState } from 'react'
import './App.css'
import BoxList from './BoxList.jsx'


const App = () => (
  <>
    <h1>Making Boxes</h1>
    <BoxList />
  </>
);

export default App
