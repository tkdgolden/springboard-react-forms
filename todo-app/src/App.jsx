import { useState } from 'react'
import TodoList from './TodoList.jsx'
import './App.css'

const App = () => (
    <>
      <h1>To Do List</h1>
      <TodoList />
    </>
);

export default App
