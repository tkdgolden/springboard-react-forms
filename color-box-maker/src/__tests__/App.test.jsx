import { describe, expect, test } from "vitest";
import { render, screen, fireEvent } from "@testing-library/react";
import App from "../App";
import { useState } from "react";

describe("App", () => {
    // smoketest
    test("renders", () => {
        render(<App />);
        const title = screen.getByText("Making", { exact: false });
        expect(title.textContent).toBe("Making Boxes");
    });

    // snapshot
    test("hasn't changed", () => {
        const {asFragment} = render(<App />);
        expect(asFragment()).toMatchSnapshot();
    });
});