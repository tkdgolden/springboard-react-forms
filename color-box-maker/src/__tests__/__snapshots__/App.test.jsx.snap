// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > hasn't changed 1`] = `
<DocumentFragment>
  <h1>
    Making Boxes
  </h1>
  <h2>
    Box List
  </h2>
  <div
    name="red"
    style="background-color: red; height: 100px; width: 100px;"
  />
  <button>
    x
  </button>
  <form>
    <label
      for="color"
    >
      Color: 
    </label>
    <input
      id="color"
      name="color"
      placeholder="color"
      type="text"
      value=""
    />
    <label
      for="height"
    >
      Height: 
    </label>
    <input
      id="height"
      name="height"
      placeholder="height"
      type="number"
      value=""
    />
    <label
      for="width"
    >
      Width: 
    </label>
    <input
      id="width"
      name="width"
      placeholder="width"
      type="number"
      value=""
    />
    <button>
      Add!
    </button>
  </form>
</DocumentFragment>
`;
